addons/stock/models/stock_lot.py

```diff
@@ -239,17 +240,24 @@ def _search_product_qty(self, operator, value):
         return [('id', 'in', ids)]
 
     def _search_partner_ids(self, operator, value):
-        domain = []
-        if isinstance(value, (int, list)):
-            domain.append(('id', operator, value))
-        elif isinstance(value, str):
-            domain.append(('name', operator, value))
-
-        if not domain:
-            return [('id', 'in', [])]
-        partners = self.env['res.partner'].search(domain)
-        filtered_lot_ids = self.env['stock.lot'].search([]).filtered(lambda lot: any(partner.id in partners.ids for partner in lot.partner_ids))
-        return [('id', 'in', filtered_lot_ids.ids)]
+        if operator in expression.NEGATIVE_TERM_OPERATORS or not isinstance(value, (Iterable)):
+            return NotImplemented
+        is_no_partner = operator == 'in' and list(value) == [False]
+        domain = Domain([
+            ('lot_id', '!=', False),
+            ('state', '=', 'done'),
+        ])
+        if is_no_partner:
+            # reverse the search, get all lots sent to partner so we can return all lots NOT sent
+            domain &= Domain('picking_partner_id', 'not in', value)
+        else:
+            domain &= Domain('picking_partner_id', operator, value)
+        domain &= Domain(self._get_outgoing_domain())
```

Comment:
We are sure that we don't also want the partner from who we received the lot/sn?